central/resourcecollection/datastore/datastore_impl.go

```diff
@@ -182,6 +222,10 @@ func (ds *datastoreImpl) AddCollection(ctx context.Context, collection *storage.
 	return err
 }
 
+func (ds *datastoreImpl) DryRunCollection(_ context.Context, collection *storage.ResourceCollection) error {
+	return ds.dryRunCollectionInGraph(collection)
```

Comment:
You must perform a permission check on `Workflow Administrator` first.

Reply:
Further explained: the dry run operation does not reach the postgres store which handles permission checks.

Reply:
addressed and added a variation of the comment